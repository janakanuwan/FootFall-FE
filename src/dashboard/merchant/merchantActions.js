import { createAction } from 'redux-actions';
import { CHANGE_MERCHANT, FETCH_MERCHANTS, SET_MERCHANTS } from '../../const/action-types';

const changeMerchant = createAction(
  CHANGE_MERCHANT,
  selectedMerchant => selectedMerchant,
);

const setMerchants = createAction(
  SET_MERCHANTS,
  merchantsResponse => merchantsResponse,
);

/**
 * async call to api {@link userFlowMiddleware}
 */
const fetchMerchants = createAction(
  FETCH_MERCHANTS,
  authInfo => authInfo,
);

export { changeMerchant, setMerchants, fetchMerchants };
