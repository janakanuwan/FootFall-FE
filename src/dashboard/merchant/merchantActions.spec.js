import { changeMerchant } from './merchantActions';

describe('merchantActions', () => {
  it('should create an action for merchant change', () => {
    expect(changeMerchant(1)).toMatchSnapshot();
  });
});


