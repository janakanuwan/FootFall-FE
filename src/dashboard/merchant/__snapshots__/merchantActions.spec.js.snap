// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merchantActions should create an action for merchant change 1`] = `
Object {
  "payload": Object {
    "id": 2,
    "name": "Test Merchant",
  },
  "type": "CHANGE_MERCHANT",
}
`;

exports[`merchantActions should create an action to fetch merchants 1`] = `
Object {
  "payload": Object {
    "email": "hello@gmail.com",
    "id": 1,
    "token": "1234",
  },
  "type": "FETCH_MERCHANTS",
}
`;

exports[`merchantActions should create an action to set failed  merchant response 1`] = `
Object {
  "error": true,
  "payload": [Error: Failed to fetch],
  "type": "SET_MERCHANTS",
}
`;

exports[`merchantActions should create an action to set merchant response 1`] = `
Object {
  "payload": Array [
    Object {
      "id": 1,
      "name": "Test merchant 1",
    },
    Object {
      "id": 2,
      "name": "Test Merchant 2",
    },
  ],
  "type": "SET_MERCHANTS",
}
`;
